Automattic/wp-calypso
# composite-checkout: Use a default registry to replace custom registry (4)

#### Changes proposed in this Pull Request

This refactor is the first step toward making it less risky to create payment methods within a component. Primarily it creates a new default `@wordpress/data` registry in the `composite-checkout` package, which is exported along with its `registerStore` function (which is part of the registry but should be more convenient in some cases). 

This registry (and its `registerStore`, `select`, and `dispatch` functions) can be imported directly by components that use the package, removing the need to create a registry and pass it throughout the component tree which, while not a bad idea, creates additional overhead for code using `composite-checkout` because it forces them to have to think about and manage a data layer even if they don't need one themselves.

Assuming this works well, it shouldn't have any notable effects and the code should continue to run as expected. After this is in place it will be easier to refactor the payment methods themselves to reduce and split up their dependencies, preventing the possibility of recreating them unnecessarily.

#### Testing instructions

- First run the demo with `npm run composite-checkout-demo` and verify that it works as it did before.
- Then run calypso and verify that everything works as it did before in composite checkout (you can use the `?flags=composite-checkout-force` query string to force the composite checkout view).
